src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1722,24 +1722,31 @@ private BoundExpression ReplaceTypeOrValueReceiver(BoundExpression receiver, boo
             {
                 case BoundKind.TypeOrValueExpression:
                     var typeOrValue = (BoundTypeOrValueExpression)receiver;
+                    typeOrValue.Data.TakeDiagnosticBagOwnership(out BindingDiagnosticBag? valueDiagnosticsBag, out BindingDiagnosticBag? typeDiagnosticsBag);
                     if (useType)
                     {
-                        diagnostics.AddRange(typeOrValue.Data.TypeDiagnostics);
+                        diagnostics.AddRange(typeDiagnosticsBag);
 
-                        foreach (Diagnostic d in typeOrValue.Data.ValueDiagnostics.Diagnostics)
+                        foreach (Diagnostic d in valueDiagnosticsBag.DiagnosticBag.AsEnumerableWithoutResolution())
                         {
-                            if (d.Code == (int)ErrorCode.WRN_PrimaryConstructorParameterIsShadowedAndNotPassedToBase &&
```

Comment:
`.Code` was the second source of recursion, once the first was addressed. It forces resolution of the diagnostic info, throwing us right back into the same problem. To address this, I added an `IsCode` method on `Diagnostic`, which will not force resolution of lazy diagnostics. The only thing lazy about lazy diagnostics is whether or not they should be reported: we do know what code they are ahead of time. Arguably, we could avoid resolving lazy diagnostics with the `Code` property. I felt that I wasn't comfortable with that big a change here, but if we'd prefer that approach, I can do that.